addons/web/static/src/js/views/calendar/calendar_model.js

```diff
@@ -273,6 +274,17 @@ return AbstractModel.extend({
         this.data.highlight_date = this.data.target_date = start.clone();
         this.data.start_date = this.data.end_date = start;
         switch (this.data.scale) {
+            case 'year':
+                const yearStart = this.data.start_date.clone().startOf('year');
+                let yearStartDay;
+                if (yearStart.day() >= this.week_start) {
+                    yearStartDay = this.week_start;
+                } else {
+                    yearStartDay = this.week_start - 7;
```

Comment:
Indeed it's [0,6]
I added the comments, it's the same with the other cases
if the start of the year is >= week_start then the start of the year is week_start
otherwise the start of the year < week_start so the start of the year is week_start of previous week

Reply:
Sorry but this isn't clear. As far as I understand:
IF case: the 1st January is after the week start (e.g. week start is Monday, 01/01 is Wednesday)
ELSE case: it isn't the case (e.g. 01/01 is Sunday and week start is Monday)

Would this be ok?
```js
let yearStartDay = this.week_start;
if (yearStart.day() < yearStartDay) {
    // the 1st of January is before our week start (e.g. week start is Monday, and
    // 01/01 is Sunday), so we go one week back
    yearStartDay -= 7;
}
```